GliaWidgetsTests/SecureConversations/Welcome/SecureConversations.WelcomeViewModelSpec.swift

```diff
@@ -0,0 +1,218 @@
+import Foundation
+import XCTest
+@testable import GliaWidgets
+
+final class SecureConversationsWelcomeViewModelTests: XCTestCase {
+    var viewModel: SecureConversations.WelcomeViewModel = .mock
+
+    override func setUp() {
+        viewModel = .mock
+    }
+
+    func testDelegateGetsCalledOnBackTapped() {
+        let expectation = expectation(description: "")
+        var isCalled = false
+
+        viewModel.delegate = { event in
+            switch event {
+            case .backTapped:
+                isCalled = true
+                expectation.fulfill()
+            default: XCTFail()
+            }
+        }
+
+        viewModel.event(.backTapped)
+
+        waitForExpectations(timeout: 1)
```

Comment:
I strongly suggest to omit waiting, to avoid flaky tests. In the SDK we have tools that allow accomplishing that, because all of the dependencies are controlled. So instead of waiting, for testing callback based results we use this approach:
```
var receivedValue: Value?

entity.methodWithCallback { value in
   receivedValue = value 
}

try XCTAssertEqual(XCTUnwrap(receivedValue), expectedValue)
```

Reply:
Removed all expectations e2b286d5e22d5ad787f913ab27d524c56417aa9e